examples/collatz/guest/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+#![cfg_attr(feature = "guest", no_std)]
+#![no_main]
+
+#[jolt::provable]
+fn collatz_convergence_range(start: u128, end: u128) -> u128 {
+    let mut max_num_steps = 0;
+    for n in start..end {
+        let num_steps = collatz_convergence(n);
+        if num_steps > max_num_steps {
+            max_num_steps = num_steps;
+        }
+    }
+    max_num_steps
+}
+
+#[jolt::provable]
+fn collatz_convergence(n: u128) -> u128 {
```

Comment:
Does this need to be marked as provable? It is not used in the host so I think we can remove it.